lib/logstash/inputs/file.rb

```diff
@@ -181,27 +182,47 @@ def register
     if @start_position == "beginning"
       @tail_config[:start_new_files_at] = :beginning
     end
+
+    if (@run_with_identity = @codec.class.config_name == 'multiline')
+      @codec = LogStash::Codecs::IdentityMapCodec.new(@codec)
+    end
```

Comment:
Given that this is not in core we can be agile about how we move from this conservative approach to the more general one you mention.  I want to limit potential problems to multiline codec usage for the near term.